apps/src/sites/studio/pages/teacher_dashboard/show.js

```diff
@@ -109,6 +111,8 @@ $(document).ready(function() {
         />
       </Router>
     </Provider>,
-    document.getElementById('teacher-dashboard')
+    document.getElementById('teacher-dashboard'),
+    () =>
+      isJavabuilderConnectionTestEnabled && testJavabuilderWebsocketConnection()
```

Comment:
i've never actually used this [ReactDOM.render](https://reactjs.org/docs/react-dom.html#render) callback before, but the react docs have a message that is slightly worrying:

> If the optional callback is provided, it will be executed after the component is rendered or updated.

does that mean this will be triggered on every update?

Reply:
Yeah that made me a bit nervous as well -- there's this other statement in the description of what ReactDOM.render does if the provided element has been rendered before:

```
If the React element was previously rendered into container, this will perform an update on it and only mutate the DOM as necessary to reflect the latest React element.
```

So I interpreted the sentence you highlighted as meaning that it will be called once, but whether that's after an initial render or an update depends on whether the component had been rendered before. I'll do another pass through to make sure that is the case though.

Reply:
Pretty confusing wording though in any case.

Reply:
Confirmed, did some navigation around the teacher dashboard without any additional websocket connections being attempted.